src/pudl/output/ferc1.py

```diff
@@ -1399,3 +1402,341 @@ def remove_intra_table_calculated_values(exploded: pd.DataFrame) -> pd.DataFrame
+    been renamed from the original reported XBRL fact name. Along with ``source_table``
+    this attribute uniquely identifies the node within the tree.
+    """
+    source_table: str
+    """The unique source table chosen for the purposes of the explosion.
+
+    Along with ``xbrl_factoid`` this attribute uniquely identifies the node within the
+    tree.
+    """
+    xbrl_factoid_original: str
+    """The original reported XBRL Fact name."""
+    weight: float
+    """The weight associated with the XBRL Fact in its source table calculation.
+
+    This metadata about the is not currently used for performing calculations.
+    """
+    children: list["XbrlCalculationTreeFerc1"] = []
+    """The subcomponents required to calculate the value of this fact, if any.
+
+    This list will be empty for leaf nodes (reported values) and for calculated values
+    that lie outside the list of tables being used in the explosion.
+    """
+    tags: dict[str, str] = {}
+    """A dictionary of categorical metadata tags associated with a node in the tree."""
+
+    def pprint(self: Self, indent=4):
+        """Print a legible JSONified version of the calculation tree."""
+        print(json.dumps(self.dict(), indent=indent))
+
+    @staticmethod
+    def _check_index(df: pd.DataFrame):
+        """Check that an input meta dataframe is appropriately indexed."""
+        idx_cols = ["table_name", "xbrl_factoid"]
+        if df.index.names != idx_cols:
+            raise AssertionError(
+                f"Exploded metadataframes must be indexed by {idx_cols}, but found "
+                f"{df.index.names=}"
+            )
+        if not df.index.is_unique:
+            raise AssertionError("Found non-unique index in exploded metadata.")
+
+    @classmethod
+    def from_exploded_meta(
+        cls,
+        source_table: str,
+        xbrl_factoid: str,
+        exploded_meta: pd.DataFrame,
+        weight: float = np.nan,
+        tags_df: pd.DataFrame | None = None,
+        propagate_weights: bool = True,
+    ) -> "XbrlCalculationTreeFerc1":
+        """Construct a complete calculation tree based on exploded metadata.
+
+        Args:
+            source_table: The table within which we are searching for the XBRL fact,
+                required as many appear in multiple tables.
+            xbrl_factoid: Equivalent to the "name" field in the calculation metadata.
+                Potentially renamed from the originally reported XBRL value.
+            exploded_meta: A dataframe containing metadata for all of the facts
+                referenced by the exploded tables. Must be indexed by the columns
+                (table_name, xbrl_factoid) and the index must be unique.
+            weight: The weight associated with this node in its calculation. Must be
+                passed in because it is only available when we have access to the
+                calculation metadata. Defaults to 1.0 so that we can create the root
```

Comment:
The meaning of the weight is different in the root node vs. all the other nodes. The root node has been plucked out of whatever its original calculation context was -- it effectively has no weight... and the only weight that really makes sense in that context is 1.0 (since presumably we're interested in the actual value that would be calculated for this node, based on all of the nested child calculations that feed into it).

In practice, for all nodes except the root node, this value is filled in with the weight which is read in from `exploded_meta` but because we need to use the same constructor for the root node and all of those child nodes which are part of the tree.  We are passing `exploded_meta` down through the recursive construction of the tree as a place to look up the weights, lists of children, etc.  I forgot to edit the docstring when I changed the default value to `np.nan`.